Microsoft.TeamServices.Samples.Client/Hooks/SubscriptionsSample.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Services.ServiceHooks.WebApi;
+using Newtonsoft.Json.Linq;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+
+namespace Microsoft.TeamServices.Samples.Client.Hooks
+{
+    [ClientSample(ServiceHooksPublisherApiConstants.AreaName, "Subscriptions")]
+    public class SubscriptionsSample : ClientSample
+    {
+        /// <summary>
+        /// Create a service hooks subscription to get notified about Git push events to a particular repo.
+        /// </summary>
+        /// <returns></returns>
+        [ClientSampleMethod]
+        public Subscription CreateGitPushWebHooksSubscription()
+        {
+            VssConnection connection = this.Context.Connection;
+            TeamProjectReference project = ClientSampleHelpers.FindAnyProject(this.Context);
+            
+            ServiceHooksPublisherHttpClient client = connection.GetClient<ServiceHooksPublisherHttpClient>();
+
+            Subscription gitPushSubscription = new Subscription()
+            {
+                PublisherId = "tfs",
+                EventType = "git.push",
+                ResourceVersion = "1.0",
+                PublisherInputs = new Dictionary<string, string>()
+                {
+                    { "projectId", project.Id.ToString() }
+                },
+                ConsumerId = "webHooks",
+                ConsumerActionId = "httpRequest",
+                ConsumerInputs = new Dictionary<string, string>()
+                {
+                    { "url", "https://someurl" }
```

Comment:
Can we point this to a https://requestb.in/ so that the user can see it happening? If so, we should include a comment describing why we did it that way.